tests/unit/test_client.py

```diff
@@ -138,3 +141,72 @@ async def test_CloudSQLClient_user_agent(
         assert client._user_agent == f"cloud-sql-python-connector/{version}+{driver}"
     # close client
     await client.close()
+
+
+async def test_cloud_sql_error_messages_get_metadata(
+    fake_credentials: Credentials,
+) -> None:
+    """
+    Test that Cloud SQL Admin API error messages are raised for _get_metadata.
+    """
+    # mock Cloud SQL Admin API calls with exceptions
+    client = CloudSQLClient(
+        sqladmin_api_endpoint="https://sqladmin.googleapis.com",
+        quota_project=None,
+        credentials=fake_credentials,
+    )
+    get_url = "https://sqladmin.googleapis.com/sql/v1beta4/projects/my-project/instances/my-instance/connectSettings"
+    resp_body = {
+        "error": {
+            "code": 403,
+            "message": "Cloud SQL Admin API has not been used in project 123456789 before or it is disabled",
+        }
+    }
+    with aioresponses() as mocked:
+        mocked.get(
+            get_url,
+            status=403,
+            payload=resp_body,
+            repeat=True,
+        )
+        with pytest.raises(ClientResponseError) as exc_info:
+            await client._get_metadata("my-project", "my-region", "my-instance")
+            assert exc_info.status == 403
+            assert (
+                exc_info.message
+                == "Cloud SQL Admin API has not been used in project 123456789 before or it is disabled"
+            )
```

Comment:
These assert statements in lines 174-178 won't be executed. I was playing around with this when implementing this change for AlloyDB: https://github.com/GoogleCloudPlatform/alloydb-python-connector/pull/400.

When an exception is raised in a `pytest.raises` block, the statement that raises the exception must be the final statement in the block. More info from the pytest documentation: 
![pdXPZFZZr6Uvykx](https://github.com/user-attachments/assets/7137c061-9b94-4801-aaa3-f953e7eddf5b)


So the assert statements should go after the block, like this:
```
with pytest.raises(ClientResponseError) as exc_info:
    await client._get_metadata(...)
assert exc_info.value.status == 404
```